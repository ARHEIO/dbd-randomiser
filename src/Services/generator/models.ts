export enum IRank {
  BROWN = 1,
  YELLOW,
  GREEN,
  PURPLE,
  IRIDESCANT,
  GOLD
}

export interface BasicObject {
  name: string;
  icon: string;
  rank: IRank | null;
}

export interface UpgradableObject extends BasicObject {
  addons: BasicObject[];
}

export type Killer = UpgradableObject; // doesn't use rank
export type Item = UpgradableObject;
export type Survivor = BasicObject;
export type Addon = BasicObject;
export type Perk = BasicObject;

export interface GeneratedKiller {
  name: string;
  icon: string;
  addons: Addon[];
  perks: Perk[];
}

export interface GeneratedSurvivor {
  icon: string;
  item: Item;
  name: string;
  perks: Perk[];
}
