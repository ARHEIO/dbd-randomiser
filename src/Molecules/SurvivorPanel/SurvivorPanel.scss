.survivor {
  display: grid;
  grid-gap: 16px;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(2, 1fr);
  padding-bottom: 32px;

  img {
    max-width: 100%;
    max-height: 140px;
  }

  & > * {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-self: center;
    border-bottom: 2px solid #FF2A2B;
    padding-bottom: 16px;
    max-width: 160px;
  }

  &_name,
  &_item,
  &_addons,
  &_perks {
    grid-column: span 1;
  }

  &_perks {
    grid-row-start: 2;
  }
}
