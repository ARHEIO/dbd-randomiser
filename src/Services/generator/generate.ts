import axios from 'axios';
import {
  GeneratedKiller, GeneratedSurvivor,
} from './models';
import config from '../../config';

export const generateKiller = (): Promise<{data: GeneratedKiller}> => axios.get(`${config.endpoints.randomiserApi}/killer`);

export const generateSurvivor = (): Promise<{data: GeneratedSurvivor}> => axios.get(`${config.endpoints.randomiserApi}/survivor`);
