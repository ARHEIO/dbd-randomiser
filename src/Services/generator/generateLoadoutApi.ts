import axios from 'axios';
import { GeneratedKiller, GeneratedSurvivor } from './models';
import config from '../../config';

export const generateKiller = (): Promise<{data: GeneratedKiller}> => axios.get(`${config.endpoints.randomiserApi}/loadout?q=killer`);

export const generateSurvivor = (): Promise<{data: GeneratedSurvivor}> => axios.get(`${config.endpoints.randomiserApi}/loadout?q=survivor`);
