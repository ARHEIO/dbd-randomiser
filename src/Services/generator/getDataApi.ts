import axios from 'axios';
import config from '../../config';
import {
  Killer, Survivor, Item, Perk,
} from './models';

export const getKiller = (id: number): Promise<{data: Killer}> => axios
  .get(`${config.endpoints.randomiserApi}/killer/${id}`);
export const getKillerWithAddons = (id: number): Promise<{data: Killer}> => axios
  .get(`${config.endpoints.randomiserApi}/killer/${id}/?expands=upgradables`);
export const getKillers = (): Promise<{data: Killer[]}> => axios
  .get(`${config.endpoints.randomiserApi}/killer`);

export const getSurvivor = (id: number): Promise<{data: Survivor}> => axios
  .get(`${config.endpoints.randomiserApi}/survivor/${id}`);
export const getSurvivors = (): Promise<{data: Survivor[]}> => axios
  .get(`${config.endpoints.randomiserApi}/survivor`);

export const getItem = (id: number): Promise<{data: Item}> => axios
  .get(`${config.endpoints.randomiserApi}/survivor/${id}`);
export const getItemWithAddons = (id: number): Promise<{data: Item}> => axios
  .get(`${config.endpoints.randomiserApi}/survivor/${id}?expands=upgradables`);
export const getItems = (): Promise<{data: Item[]}> => axios
  .get(`${config.endpoints.randomiserApi}/survivor`);

export const getSurvivorPerk = (id: number): Promise<{data: Perk[]}> => axios
  .get(`${config.endpoints.randomiserApi}/perk/survivor/${id}`);
export const getSurvivorPerks = (): Promise<{data: Perk}> => axios
  .get(`${config.endpoints.randomiserApi}/perk/survivor`);

export const getKillerPerk = (id: number): Promise<{data: Perk}> => axios
  .get(`${config.endpoints.randomiserApi}/perk/killer/${id}`);
export const getKillerPerks = (): Promise<{data: Perk[]}> => axios
  .get(`${config.endpoints.randomiserApi}/perk/killer`);
