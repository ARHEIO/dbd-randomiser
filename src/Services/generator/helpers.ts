export const randomNumberGenerator = (max: number) => {
  return Math.floor(Math.random() * max)
}

export const getMultiple = (arrayOfThings: any[], numberToGet: number) => {
  return arrayOfThings && arrayOfThings.length
    ? arrayOfThings
        .map(x => ({ x, r: Math.random() }))
        .sort((a, b) => a.r - b.r)
        .map(a => a.x)
        .slice(0, numberToGet)
    : [];
}

export const getAddons = (addons: any[]) => getMultiple(addons, 2)

export const getPerks = (perks: any[]) => getMultiple(perks, 4)