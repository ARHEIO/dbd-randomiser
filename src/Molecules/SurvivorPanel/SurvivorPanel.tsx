import React, { ReactElement } from 'react';
import './SurvivorPanel.scss';

import DiamondIcon from '../../Atoms/DiamondIcon/DiamondIcon';
import SquareIcon from '../../Atoms/SquareIcon/SquareIcon';

import { Perk as IPerk, GeneratedSurvivor } from '../../Services/generator/models';

const SurvivorPanel = (props: {survivor: GeneratedSurvivor}): ReactElement => {
  const { survivor } = props;
  return (
    <div className="survivor">
      <SquareIcon name={survivor.name} icon={survivor.icon} rank={null} />
      <SquareIcon name={survivor.item.name} icon={survivor.item.icon} rank={survivor.item.rank} />
      {survivor.item.addons.length > 0
        ? survivor.item.addons.map((addon: any) => (
          <SquareIcon key={addon.name} name={addon.name} icon={addon.icon} rank={addon.rank} />
        ))
        : <div className="firecracker-buffer" />}
      {survivor.perks.map((perk: IPerk) => (
        <DiamondIcon key={perk.name} name={perk.name} icon={perk.icon} rank={perk.rank} />
      ))}
    </div>
  );
};
export default SurvivorPanel;
