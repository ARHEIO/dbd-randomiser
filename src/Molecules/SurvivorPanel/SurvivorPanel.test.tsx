import React from 'react';

import {
  render,
} from '@testing-library/react';
import pretty from 'pretty';
import SurvivorPanel from './SurvivorPanel';
import { GeneratedSurvivor } from '../../Services/generator/models';

const defaultObject: GeneratedSurvivor = {
  name: 'Adam',
  icon: 'assets/adam.png',
  item: {
    addons: [
      { icon: '', name: 'Evoluent Vertical Mouse 4', rank: 5 },
      { icon: '', name: 'OLKB Preonic', rank: 5 },
    ],
    name: 'MacBook Pro 2018',
    rank: 5,
    icon: '',
  },
  perks: [
    { icon: '', rank: 5, name: 'Archetecture' },
    { icon: '', rank: 5, name: 'Dockerised' },
    { icon: '', rank: 5, name: 'Carpel Tunneling' },
    { icon: '', rank: 5, name: '8 Hour Nap' },
  ],
};

describe('Search View', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('should render when the default object is passed', () => {
    const { container } = render(<SurvivorPanel survivor={defaultObject} />);
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it('should render when no addons are passed', () => {
    const addonlessObject = defaultObject;
    addonlessObject.item.addons = [];
    const { container } = render(<SurvivorPanel survivor={addonlessObject} />);
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
