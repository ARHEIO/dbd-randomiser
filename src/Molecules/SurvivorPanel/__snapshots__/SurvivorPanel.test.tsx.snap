// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search View should render when no addons are passed 1`] = `
"<div class=\\"survivor\\">
  <div class=\\"square-icon\\">
    <p class=\\"square-icon_name\\">Adam</p>
    <div class=\\"square-icon_icon\\" style=\\"background-color: rgb(97, 81, 60);\\"><img alt=\\"Adam\\" src=\\"https://dqr3pglopijar.cloudfront.net/assets/adam.png\\"></div>
  </div>
  <div class=\\"square-icon\\">
    <p class=\\"square-icon_name\\">MacBook Pro 2018</p>
    <div class=\\"square-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"MacBook Pro 2018\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
  </div>
  <div class=\\"firecracker-buffer\\"></div>
  <div class=\\"diamond-icon\\">
    <p class=\\"diamond-icon_name\\">Archetecture</p>
    <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"Archetecture\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
  </div>
  <div class=\\"diamond-icon\\">
    <p class=\\"diamond-icon_name\\">Dockerised</p>
    <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"Dockerised\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
  </div>
  <div class=\\"diamond-icon\\">
    <p class=\\"diamond-icon_name\\">Carpel Tunneling</p>
    <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"Carpel Tunneling\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
  </div>
  <div class=\\"diamond-icon\\">
    <p class=\\"diamond-icon_name\\">8 Hour Nap</p>
    <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"8 Hour Nap\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
  </div>
</div>"
`;

exports[`Search View should render when the default object is passed 1`] = `
"<div class=\\"survivor\\">
  <div class=\\"square-icon\\">
    <p class=\\"square-icon_name\\">Adam</p>
    <div class=\\"square-icon_icon\\" style=\\"background-color: rgb(97, 81, 60);\\"><img alt=\\"Adam\\" src=\\"https://dqr3pglopijar.cloudfront.net/assets/adam.png\\"></div>
  </div>
  <div class=\\"square-icon\\">
    <p class=\\"square-icon_name\\">MacBook Pro 2018</p>
    <div class=\\"square-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"MacBook Pro 2018\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
  </div>
  <div class=\\"square-icon\\">
    <p class=\\"square-icon_name\\">Evoluent Vertical Mouse 4</p>
    <div class=\\"square-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"Evoluent Vertical Mouse 4\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
  </div>
  <div class=\\"square-icon\\">
    <p class=\\"square-icon_name\\">OLKB Preonic</p>
    <div class=\\"square-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"OLKB Preonic\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
  </div>
  <div class=\\"diamond-icon\\">
    <p class=\\"diamond-icon_name\\">Archetecture</p>
    <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"Archetecture\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
  </div>
  <div class=\\"diamond-icon\\">
    <p class=\\"diamond-icon_name\\">Dockerised</p>
    <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"Dockerised\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
  </div>
  <div class=\\"diamond-icon\\">
    <p class=\\"diamond-icon_name\\">Carpel Tunneling</p>
    <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"Carpel Tunneling\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
  </div>
  <div class=\\"diamond-icon\\">
    <p class=\\"diamond-icon_name\\">8 Hour Nap</p>
    <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(255, 9, 85);\\"><img alt=\\"8 Hour Nap\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
  </div>
</div>"
`;
